File: tme5/writer_reader/writer.py
#!/usr/bin/env python
import os, time

pipe_name = '/tmp/myfifo'

if not os.path.exists(pipe_name):
    os.mkfifo(pipe_name)

pipe_out = open(pipe_name,'w')

i=0;
while i < 5:
    pipe_out.write("hello %d fois from python\n" % (i+1,))
    pipe_out.flush()
    time.sleep(1)
    i=i+1

pipe_out.write("end\n")

File: tme5/writer_reader/writer.c
#include <fcntl.h>
#include <sys/stat.h>
#include <sys/types.h>
#include <unistd.h>
#include <stdio.h>

int main()
{
	int		fd;
	char    *myfifo = "/tmp/myfifo";

	/* create the FIFO (named pipe) */
	mkfifo(myfifo, 0666);

	/* write "Hi" to the FIFO */
	fd = open(myfifo, O_WRONLY);
    for(int i=0; i<5; i++) {
        char    buf[80];
        int len=sprintf(buf, "hello %d fois from c\n", i+1);
		write(fd, buf, len);
		sleep(1);
	}
	write(fd, "end\n", sizeof("end"));
	close(fd);

	/* remove the FIFO */
	unlink(myfifo);

	return 0;
}
